### ISSUE TITLE
System.Drawing.Imaging.Metafile(filename) throws wrong Exception when filename is not valid on netcoreapp

### ISSUE THREAD

**Issue by KostaVlev** [dotnet/corefx#22640]
**Test:**

```c#
[ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
public void Ctor_InvalidPath_ThrowsArgumentException()
{
    string fileNameTooLong = new string('a', 261);
    AssertExtensions.Throws<ArgumentException>(null, () => new Metafile(string.Empty)); // fails on netcoreapp
    AssertExtensions.Throws<ArgumentException>(null, () => new Metafile(@"fileNo*-//\\#@(found")); // fails on netcoreapp
    AssertExtensions.Throws<ArgumentException>(null, () => new Metafile(fileNameTooLong)); // fails on netcoreapp
}
```

Expected:  `System.ArgumentException`

Actual:
netfx `System.ArgumentException`
netcoreapp `typeof(System.Runtime.InteropServices.ExternalException): A generic error occurred in GDI+`

(#20711)

**Comment by mellinoe** [comment]
This is another case where `Path.GetFullPath` needs to be called. http://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/Advanced/Metafile.cs,112

Fixing this along with the other items I've mentioned.

**Comment by KostaVlev** [comment]
Fixed by #22941
